Domain change: Domain purchase success screen

0nko opened the pull request:
Fixes #8366, a subtask of #8238.

The PR just adds the domain purchase success screen, which displays the selected domain. There was a problem with exporting a vector image from Figma (I asked Ann if she could fix it) so in the meantime I used a PNG as a placeholder.

<img src="https://user-images.githubusercontent.com/1522856/220458196-9c26cbe2-738b-45f6-a60d-41947856d725.png" width="300" />

0nko committed:
Rename the domain dashboard screen file

0nko committed:
Add string and image resources

0nko committed:
Pass the purchased domain to the success screen using navigation

0nko committed:
Add the new purchase successful screen and ViewModel

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/domain/PurchaseSuccessfulScreen.kt:
```diff
@@ -0,0 +1,127 @@
+
+@Composable
+private fun PurchaseSuccessful(viewState: ViewState, onDoneButtonClicked: () -> Unit) {
+    Column(modifier = Modifier.background(MaterialTheme.colors.surface)) {
```
Np, I'd make the content scrollable. At least the part above the "Done" button. If you rotate the screen you can't see anything.

0nko replied:
Good call! 👍

JorgeMucientes commented:
Looks good @0nko, nice one! I left a minor non blocking np. 

On other side of things I was wondering if we should provide a different illustration for dark mode as we do on other screens. I think the current one looks good in dark mode though (once we update it to vector). 
cc @annchichi 

<img width="300"  src="https://user-images.githubusercontent.com/2663464/220945540-171d6e91-abd8-4d33-a97b-06212c8a923d.png"> 

Btw friendly reminder (in case you haven't addressed it already), we should update the color of the "Primary" domain tag for dark mode: 

<img width="300"  src="https://user-images.githubusercontent.com/2663464/220945523-ac51aeb9-3d55-47cc-9d9a-c63c24fcefcb.png">

0nko committed:
Replace the domain success image with a vector image

0nko committed:
Make the domain purchase screen scrollable

0nko committed:
Add dark mode color variants for the primary domain tag

0nko commented:
Thank you for the review, @JorgeMucientes. I've addressed your comments, including the primary domain tag:

![image](https://user-images.githubusercontent.com/1522856/221021841-9a2b60c8-e49a-49ed-a3a9-be352ccadca0.png)

I've left the illustration as is since we don't have a dark variant and I also think it looks good.
